lambdas/ec2/hipchat_notifier.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+import boto3
+import json
+import ast
+import zlib
+import base64
+import os
+from urllib2 import Request, urlopen
+
+HIPCHAT_TOKEN = 'LOCAL_TOKEN'
+HIPCHAT_ROOM_ID = 'LOCAL_ID'
+
+def get_account_alias():
+    """
+    Return the first alias listed from Amazon.
+    """
+    client = boto3.client('iam')
+    return client.list_account_aliases()['AccountAliases'][0]
+
+def read_token():
+    """
+    Read in the environment HIPCHAT_TOKEN, or read the local token.
+    """
+
+    if 'HIPCHAT_TOKEN' in os.environ:
+        return os.environ['HIPCHAT_TOKEN']
+    else:
+        return HIPCHAT_TOKEN    
+
+def read_room_id():
+    """
+    Read in the environment HIPCHAT_ROOM_ID, or read in the local id.
+    """
+    if 'HIPCHAT_ROOM_ID' in os.environ:
+        return int(os.environ['HIPCHAT_ROOM_ID'])
+    else:
+        return HIPCHAT_ROOM_ID
+
+def process_subscription_notification(event):
+    """
+    param: event: AWS log event.
+
+    Decompresses the data from an AWS Log Event and returns a standard dict.
+    """
+
+    zipped = base64.standard_b64decode(event['awslogs']['data'])
+    unzipped_string = zlib.decompress(zipped, 16+zlib.MAX_WBITS)
+    event_dict = ast.literal_eval(unzipped_string)
+    return event_dict
+    
+def post(event, context):
+    """
+    :param event: AWS Log Event.
+    :param context: Object to determine runtime info of the Lambda function.
+
+    See http://docs.aws.amazon.com/lambda/latest/dg/python-context-object.html for more info
+    on context.
+    Process an AWS Log event and post it to a Hipchat Room.
+    """
+
+    V2TOKEN = read_token()
+    ROOMID = read_room_id()
+    url = 'https://api.hipchat.com/v2/room/%d/notification' % ROOMID
+
+    event_processed = process_subscription_notification(event)
+    
+    for log_event in event_processed['logEvents']:
+
+        message = log_event['message']
+        headers = {
+            "content-type": "application/json",
+            "authorization": "Bearer %s" % V2TOKEN}
+        datastr = json.dumps({
+            'message': message,
+            'color': 'purple',
+            'message_format': 'html',
+            'notify': False,
+            'from': get_account_alias()})
+        request = Request(url, headers=headers, data=datastr)
+        uo = urlopen(request)
```

Comment:
@tvpartytonight not a blocking change, but if you're already making other changes, making this variable name more descriptive would be great.